design/proposals/spike-cli-dockerless.md

```diff
@@ -0,0 +1,128 @@
+  * [Background](#background)
+      - [Requirements](#requirements)
+  * [Research](#research)
+      - [Estimation](#estimation)
+      - [Takeways and recommendation](#takeways-and-recommendation)
+  * [Documentation](#documentation)
+  * [Open questions](#open-questions)
+  * [Implementation plan](#implementation-plan)
+
+## Useful links
+
+[//]: # "Add links that may be useful for the reader"
+
+| **Term**   | **Description**                                              |
+| ---------- | ------------------------------------------------------------ |
+| Epic       | https://app.zenhub.com/workspaces/dap-master-no-docker-5f82b33fdb531f00128c0a07/issues/cyberark/conjur/1892 |
+| Ruby CLI   | https://github.com/cyberark/conjur-cli                       |
+| Golang CLI | https://github.com/cyberark/conjur-cli-go                    |
+| Python CLI | https://github.com/cyberark/conjur-api-python3               |
+
+## Background
+
+[//]: # "Give relevant background for the designed feature. What is the motivation for this solution?"
+
+This document is one piece in the bigger piece of the Dockerless Conjur puzzle. We need to be able to deploy Conjur on a VM. This means we will need to figure out a way to deliver a containerless CLI to allow human-users to interface with Conjur.
+
+In this document we will discuss possible options for a containerless CLI and present a decision for which one we will pursue.
+
+#### Requirements
+
+The CLI design should answer the following requirements:
+
+*Core functionality*
+
+1. For each platform, we should support *init*, *login*, *authenticate*, *load policy*, and *fetch secrets* functionalities. These functionalities are must-haves but we should not be limited to providing just these.
+
+*Deployment*
+
+1. Provide a manual and automated way to deploy the CLI
+2. Require a maximum of 2 steps
+3. ***Recommended:*** cross-platform deployment (Linux, Windows) should all have the same flow
+
+## Research
+
+[//]: # "Elaborate on the issue you are writing a solution for"
+
+Currently, we provide 3 CLIs that can be used to interface with Conjur, Ruby, Golang, Python. We will explore and differentiate each in terms of their functionality, quality, UX, testing, security and what it is required to deploy them.
+
+|            | Functionality                                                |
+| ---------- | ------------------------------------------------------------ |
+| Ruby CLI   | Most expressive coverage                                     |
+| Golang CLI | Implementation is limited with poor architectural design. We could salvage what is currently in-place but it was recommended by the contributor to rethink the architecture and even perform a complete rewrite. |
+| Python CLI | More or less at level of Golang CLI in terms of functionality coverage but with more intuitive commands. For example, in Golang `conjur-cli variable value <variable_id>` and in Python `conjur-cli variable get <variable_id>` . |
```

Comment:
@sigalsax can you please specify all the UX differences?
@alexkalish was this UX been reviewed by PO ? by PM?

Reply:
> I was under the impression that any CLI we develop, we'd keep the same commands as they are. Otherwise, we could potentially break functionality of existing automation that uses the Ruby CLI.

I agree about your point here but see my other response on Ruby CLI un-intuitiveness as to why the persistent deficiencies of our Ruby CLI have been a good catalyst to have this improved.